lua/rzls/server/methods/codeactionresolve.lua

```diff
@@ -0,0 +1,27 @@
+local documentstore = require("rzls.documentstore")
+local razor = require("rzls.razor")
+
+---@param params lsp.CodeAction
+---@return lsp.CodeAction | nil
+return function(params)
+    local rvd = documentstore.get_virtual_document(params.data.TextDocument.uri, razor.language_kinds.razor)
+
+    if not rvd then
+        return
+    end
+
+    -- HACK: neovim does not deserialize `null`s and rzls expects an explicit `null` value
+    -- in the `params.data.delegatedDocumentUri` field.
+    params.data.delegatedDocumentUri = vim.NIL
```

Comment:
Can we build this into a variable of `required_fields` and then `deep_extend`. Just so we don't overwrite if there is something in this field in future